numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -353,6 +353,21 @@ gentype_repr(PyObject *self)
     return ret;
 }
 
+static PyObject *
+genint_type_str(PyObject *self)
+{
+    PyObject  *item, *item_str;
+    item = gentype_generic_method(self, NULL, NULL, "item");
+    if (item == NULL) {
+        return NULL;
+    }
+
+    item_str = PyObject_Str(item);
```

Comment:
Now that this is much simpler, can we add a `genint_type_repr` that does the same thing but calls `PyObject_Repr`, for completeness?